src/middlewares/schema.ts

```diff
@@ -83,6 +84,10 @@ export function checkSchema(
         // Using "!" because typescript doesn't know it isn't undefined.
         const methodCfg = config[method]!;
 
+        if (method === "if") {
+          return methodCfg ? chain : null;
+        }
+        
```

Comment:
I would add this check after the options handling so that everything is ordered correctly. Note that we have to add it before the `not` check as it would be run before it.
It should also be replaced like this so that it is type checked by TS:
```ts
if (isValidatorOptions(method, methodCfg) && methodCfg.if) {
  chain.if(methodCfg);
}
```